### ISSUE TITLE
Nested routers do not include (?P<format>) in regex

### ISSUE THREAD

**Issue by Symmetric** [alanjds/drf-nested-routers#70]
Here's an example excerpt of some normal ViewSet URL from Django 1.9.1:

```
^ ^users/$ [name='user-list']
^ ^users\.(?P<format>[a-z0-9]+)/?$ [name='user-list']
```

This allows the URL `/users.csv/` if a CSV formatter has been installed.

An equivalent url generated by drf-nested-routers:

```
^ ^users/(?P<user_pk>[^/.]+)/accounts/$ [name='user-account-list']
```

There is no entry containing the (?P<format>) regex that is required to support view formatters.

Is this a known missing feature, a bug, or a misconfiguration on my part? It's not clear from my reading of the docs how to get the format URLs.

**Comment by alanjds** [comment]
Hi.

I really dont know. I had not looked at the code, but remember to just pass
one DRF router into another, so is not up to NestedRouters to do the job.

Please let us know if I am wrong.
Thanks.

**Comment by Symmetric** [comment]
Ok, here's the problem. The DRF DefaultRouter class handles adding the .format variants in its `get_urls()` method (https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/routers.py#L320).

Because `NestedSimpleRouter`s are distinct trees of URLs that are included separately, this final format_suffix_patterns step is not executed.

I can see two possible solutions -- 

1) add
```
if self.include_format_suffixes:
            urls = format_suffix_patterns(urls)
```
to the `NestedSimpleRouter`'s `get_url()` method. Not very generic, but looks like it will work. Sketch of this is here: https://github.com/alanjds/drf-nested-routers/pull/71

2) Come up with a cunning way to arrange for `default_router.urls` to include the URLs from nested routers that have been attached under it; this would mean that you'd no longer do 
```
urlpatterns = patterns('',
    url(r'^', include(router.urls)),
    url(r'^', include(domains_router.urls)),
)
```
and instead the first include() would add all the URLs. Then the `DefaultRouter` would be responsible for generating the format suffixes on the nested routers' URLs.

2) sounds a bit more tricky, but potentially more robust to future changes in upstream code; if you have any thoughts on that approach I'd be interested to hear. That said, looks like my quick solution 1) above has done the trick, so I may just stick with that (or subclassing NestedSimpleRouter if that patch isn't accepted upstream)

**Comment by alanjds** [comment]
The 2nd solution is quite nice, and par with one of @tomchristie's proposal of final APIs. But is tricky. If you got how to do it, I am ok with that

I got your PR on #71, but Travis liked it not. Can you please fix the tests?

**Comment by alanjds** [comment]
Just released v0.90.0 including #91. This should had fixed this one. Please reopen if yet broken.
